app/src/main/java/io/github/sheasmith/discountdetective/ui/screens/ShoppingListScreen.kt

```diff
@@ -165,7 +173,10 @@ private fun ProductList(
                     val elevation = animateDpAsState(if (isDragging) 16.dp else 0.dp)
                     ProductCard(
                         product = product,
-                        elevation = elevation
+                        elevation = elevation,
+                        checked = product.third.checked,
+                        onCheckedChanged = { checked -> onCheckedItem(product.third, checked) },
```

Comment:
Keep the callbacks here though. This is the correct use of them.

Reply:
Just change it so that `product.third.checked` is set here instead, plus the viewmodel save function is called.

Reply:
I get what you mean but not to sure how to do this